### DISCUSSION TITLE
how to pass the public parameter builders.

### DISCUSSION THREAD

**Discussion by demeus** [rappasoft/laravel-livewire-tables#683] [General]
Passing a variable.

` <livewire:user.table  type="user" />`

> DataTableComponent

`<?php 
   public string $type;

 public function mount($type)
    {
        $this->type     = $type;
    }

 public function builder(): Builder
    {
       User::query()
->where('signup', $this->type === 'user')
    });
    }
`
But I get this error.
**Typed property App\Http\Livewire\User\Table::$type must not be accessed before initialization**

**Comment by invaders-xx** [comment]
just try public string $type='';

**Reply by rappasoft** [reply]
Yes, this is a common Livewire error (it's not a bug).

**Reply by demeus** [reply]
Well, what about dynamic parameters?

` <livewire:user.table type="user" :platform='$platform' />`

`<?php
class UserTable extends DataTableComponent
{
public string $type='';

public Platform $platform;

public function mount(Platform $platform, $type)
{
        $this->type = $type;
        $this->platform = $platform;
}

public function builder(): Builder
{
User::query()
   ->where('signup', $this->type === 'user')
    ->whereRelation('platform', 'platform_id', $this->platform->id)
});
}
`

Error 
**_Typed property App\Http\Livewire\User\Table::$platform must not be accessed before initialization_**
